### PR TITLE
Add feature support contain and cover options on Texture (#29)

### PR DESCRIPTION
**Pull request by Drulokia**

### CODE REVIEW

File: src/core/CoreNode.ts [hunk 1]
```diff
@@ -319,8 +319,39 @@ export class CoreNode extends EventEmitter implements ICoreNode {
     }
   }
 
+  applyResizeMode(dimensions: Dimensions) {
```
Comment (frank-weindel) [inline src/core/CoreNode.ts:322]: This feature should only affect the texture coordinates that are mapped to the quad that is rendering the texture. Not actually manipulate the CoreNode's properties. Basically these variables in addQuad() of the WebGlCoreRenderer need to be manipulated depending on what is set here in the resizeMode options.

```
    let texCoordX1 = 0;
    let texCoordY1 = 0;
    let texCoordX2 = 1;
    let texCoordY2 = 1;
```


The implementation of SubTexture right below the variable declarations actually is in the ballpark of how this feature needs to be implemented.
```
    if (texture instanceof SubTexture) {
      const { x: tx, y: ty, width: tw, height: th } = texture.props;
      const { width: parentW = 0, height: parentH = 0 } = texture.parentTexture
        .dimensions || { width: 0, height: 0 };
      texCoordX1 = tx / parentW;
      texCoordX2 = texCoordX1 + tw / parentW;
      texCoordY1 = ty / parentH;
      texCoordY2 = texCoordY1 + th / parentH;
      texture = texture.parentTexture;
    }
```

Happy to hop on a call to discuss further

File: src/core/CoreTextureManager.ts [hunk 2]
```diff
@@ -55,6 +55,20 @@ export interface TextureManagerDebugInfo {
   idCacheSize: number;
 }
 
+export type resizeModeOptions =
```
Comment (frank-weindel) [inline src/core/CoreTextureManager.ts:58]: Types should always start with a capital letter

File: src/core/CoreTextureManager.ts [hunk 3]
```diff
@@ -127,6 +141,33 @@ export interface TextureOptions {
...
+   * You can clip part of a texture with clipX and clipY which are values from 0 to 1.
+   *
+   * To clip the top, set clipY to 0. To clip the bottom, set clipY to 1.
+   * By default, (clipX, clipY) is set to (0.5, 0.5). This causes an equal amount to
+   * be clipped away from left/right or top/bottom.
+   *
+   * Resize mode 'contain' ensures that an image fits in a specific rectangular area.
+   * One axis will fit to the specified size, while the other will be less than specified.
+   * @param {Object} props
+   * @param {string} props.type - The type of resize mode to use
```
Comment (frank-weindel) [inline src/core/CoreTextureManager.ts:164]: Documentation for the options can be provided directly in the above type for each property

File: examples/tests/resize-mode.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+import {
+  type INode,
+  type Dimensions,
+  type NodeLoadedEventHandler,
+} from '@lightningjs/renderer';
+import rockoImg from '../assets/rocko.png';
+import environmentImg from '../assets/robot/environment.png';
+import type { ExampleSettings } from '../common/ExampleSettings.js';
+
+export default async function ({
```
Comment (frank-weindel) [inline examples/tests/resize-mode.ts:29]: Please make this into a Visual Regression Test

**Review by wouterlucas — APPROVED** [review]

**Pull request merged**